### PR TITLE
ER Diagraman and DB implementation in rails

### PR DESCRIPTION
**Pull request by diegotc86**
Close #10 

The proposal ERD is the following:
Time Tracker ERD

On rails the following has been done:
- Generate models and create migrations (`rails g model ....`)
- Create DB (`rails db:create`)
- Run migrations (`rails db:migrate`)

### CODE REVIEW

File: rails-time-track/app/models/project.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Project < ApplicationRecord
```
Comment (Gatts1) [inline rails-time-track/app/models/project.rb]: You should add the relation with other tables. For instances  `has_many :WeeklyProjectReport`

File: rails-time-track/db/migrate/20190709220613_create_projects.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class CreateProjects < ActiveRecord::Migration[5.2]
+  def change
+    create_table :projects do |t|
+      t.string :name
+      t.string :client
+      t.string :category
+      t.date :start
+      t.date :end
```
Comment (Gatts1) [inline rails-time-track/db/migrate/20190709220613_create_projects.rb]: I think you can't define a attribute with a reserve word like `end`, maybe you can try with `endProjected` or something like that.

File: rails-time-track/app/models/project.rb (hunk 1 above)
Comment (diegotc86) [reply rails-time-track/app/models/project.rb]: Yeah, you are right. I have added all the has_many and has_many through necessary

File: rails-time-track/db/migrate/20190709220613_create_projects.rb (hunk 2 above)
Comment (diegotc86) [reply rails-time-track/db/migrate/20190709220613_create_projects.rb]: Thanks, I have modified to end_date and start_date

**Pull request merged**